### PR TITLE
Refs #24121 -- Added__repr__() to StaticNode

### PR DESCRIPTION
**Pull request by saeedblanchette**

### CODE REVIEW

File: django/templatetags/static.py [hunk 1]
```diff
@@ -98,6 +98,13 @@ def __init__(self, varname=None, path=None):
         self.path = path
         self.varname = varname
 
+    def __repr__(self) -> str:
+        return "<%s path='%s' varname='%s' >" % (
```
Comment (pope1ni) [inline django/templatetags/static.py:102]: If you look at `__init__`, this is a simple class that is easy to reconstruct, so we don't need to use the `<…>` pattern. Also consider that `varname` could be `None` so shouldn't be manually quoted like this, get it's repr instead. I'd change this to the following:

```python
        return f'{self.__class__.__name__}(path={self.path!r}, varname={self.varname!r})'
```

File: tests/template_tests/test_templatetags.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from django.templatetags.static import StaticNode
+from django.test import SimpleTestCase
+
+
+class StaticNodesTest(SimpleTestCase):
```
Comment (pope1ni) [inline tests/template_tests/test_templatetags.py:5]: I'm sure there must be existing tests for `StaticNode` in another file where these can be moved?

Comment (saeedblanchette) [reply tests/template_tests/test_templatetags.py:5]: I searched through the Django codebase and I didn't find any

Comment (pope1ni) [reply tests/template_tests/test_templatetags.py:5]:
```suggestion
class StaticNodeTests(SimpleTestCase):
```

Please can you move this test case into `tests/template_tests/syntax_tests/test_static.py`.

**Review by pope1ni — COMMENTED** [review]
Thanks for the updates. I had a look around for the most suitable place for the test, see comment below.

File: tests/template_tests/syntax_tests/test_static.py [hunk 3]
```diff
@@ -69,3 +70,17 @@ def test_static_statictag_without_path(self):
         msg = "'static' takes at least one argument (path to file)"
         with self.assertRaisesMessage(TemplateSyntaxError, msg):
             self.engine.render_to_string('t')
+
+
+class StaticNodesTest(SimpleTestCase):
```
[the remark this answers is not in the document]
Comment (saeedblanchette) [reply tests/template_tests/syntax_tests/test_static.py:75]: Sorry. I'm about to change it  Thank you for reviewing

File: django/templatetags/static.py (hunk 1 above)
Comment (saeedblanchette) [reply django/templatetags/static.py]: thank you for reviewing,  I'm not sure that I get the last point about `varnam` . do you mean? I should use `repr() `function on `varnam`

**Comment by felixxm** [comment]
@saeedblanchette Thanks :+1: Welcome aboard :boat:

**Pull request merged**